metaci/build/models.py

```diff
@@ -701,21 +700,19 @@ def record_result(self):
     def load_test_results(self):
         has_results = False
 
-        root_dir_robot_path = f"{self.root_dir}/output.xml"
-        # Load robotframework's output.xml if found
-        if os.path.isfile("output.xml"):
+        root_dir_robot_path = Path(f"{self.root_dir}/output.xml")
+        if Path("output.xml").is_file():
+            # robot test results already created after task execution
+            # see: metaci.build.flows.MetaCIFlowCallback.post_task()
             has_results = True
-            import_robot_test_results(self, "output.xml")
 
-        elif os.path.isfile(root_dir_robot_path):
+        elif root_dir_robot_path.is_file():
             # FIXME: Not sure why robot stopped writing into the cwd
             # (build temp dir) but this should handle it so long as
             # only one build runs at a time
             has_results = True
-            try:
-                import_robot_test_results(self, root_dir_robot_path)
-            finally:
-                os.remove(root_dir_robot_path)
+            # import_robot_test_results(self, root_dir_robot_path)
+            root_dir_robot_path.unlink()
```

Comment:
I believe this whole section (lines 703-715) can go away.

Removing output.xml should be done in the same place where the robot results are imported (where we know for a fact where it is).

As we discussed the other day, we don't need `has_results` at all. We can simply remove the condition on line 749 and always do those queries to count the results.